### ISSUE TITLE
Occasional validation error (buffer offset out of bounds)

### ISSUE THREAD

**Issue by andrei-drexler** [Novum/vkQuake#307]
While running with validation enabled I've noticed this error pop up a few times:
> Validation Error: [ VUID-vkCmdBindVertexBuffers-pOffsets-00626 ] Object 0: handle = 0x4b4630000000261, name = progs/v_shot.mdl, type = VK_OBJECT_TYPE_UNKNOWN; | MessageID = 0x4080ede | vkCmdBindVertexBuffers() offset (0x1800) is beyond the end of the buffer. The Vulkan spec states: All elements of pOffsets must be less than the size of the corresponding element in pBuffers (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdBindVertexBuffers-pOffsets-00626)

It took a while to figure out, but the easiest way to reproduce it is to switch to the shotgun when swinging the axe *diagonally* (vertical swing doesn't trigger it):

https://user-images.githubusercontent.com/5121829/122653402-d4a16a00-d14c-11eb-9a3c-76a277fea578.mp4

After some digging, it appears to be caused by an out of bounds `previouspose` value, so I've added some checks in R_AliasSetupFrame to at least make sure the error doesn't propagate. Disabling lerping in V_CalcRefdef after viewmodel changes fixes this entirely (since previouspose/currentpose get reset), but I'm not familiar with the entity-handling part of the code, though, so maybe there's a better way to deal with this.

P.S. I've also narrowed down the issue to commit d9d4be488436c50afcc75b2aacdb50127f6546ab, if it helps in any way.

**Comment by Novum** [comment]
Thanks for the report and proposed fix. I'll look at it asap.

**Comment by Novum** [comment]
Also good to know that it's from the QSS merge. Might be better to fix it outside the rendering code.

**Comment by Novum** [comment]
@andrei-drexler Does this happen for you if you run with `cl_nopext 1`? I suspect not.

**Comment by andrei-drexler** [comment]
You're right, I can't reproduce it with cl_nopext 1.

**Comment by Novum** [comment]
Right, that's what I was afraid of. I merged the QSS protocol stuff to fix some issue with stuttering on elevators and for the CSCQ merge, but it was a lot of code and I didn't fully trust it.

If I can't debug it I might need to take it out again.

**Comment by andrei-drexler** [comment]
I hope it doesn't come to that, the code seems in pretty good shape considering the sheer volume of these changes. I've played through all of alkaline with in-progress builds, and the only thing that I've noticed during that time was the short-lived savegame bug.

As for the current issue, I've looked into it a bit more, and it seems that the reason it doesn't happen with `cl_nopext 1` is this bit of code in `CL_ParseClientdata`, which doesn't get called when using the FTE protocol and apparently doesn't have an equivalent in that code path:

https://github.com/Novum/vkQuake/blob/312d3d8ec41fa341862ce865b7182a4fa172103e/Quake/cl_parse.c#L1481-L1486

Moving that check from `CL_ParseClientdata` to the end of `CL_RelinkEntities` in `cl_main.c` (called regardless of protocol) looks like a reasonable solution.
